src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -31,8 +31,22 @@ public abstract partial class JsonTypeInfo
         private Action<object>? _onDeserialized;
 
         /// <summary>
-        /// Object constructor. If set to null type is not deserializable.
+        /// Gets or sets a parameterless constructor to be used on deserialization.
```

Comment:
I find it confusing that this is called a "constructor".  I suggest using "factory" or something like that.  These aren't ctors in the .NET sense.